### DISCUSSION TITLE
Import Solution with SkipLowerVersion and HoldingSolution => guidance?

### DISCUSSION THREAD

**Discussion by YannickRe** [microsoft/powerplatform-build-tools#978] [Q&A]
My goal is to setup a generic deployment pipeline that can handle both the import of a new solution (not pre-existing in the target environment) and upgrading of an existing solution with a new version.
Currently, I can't seem to figure it out and wonder if there is guidance around this.

**Example 1**
- Actions Import Solution (Skip Lower Version false & Holding Solution true) and Apply Solution Upgrade in pipeline
Result: this works for existing solutions (if version is higher). This fails for new solutions with the message it can't import as a holding when there isn't an existing solution installed.

**Example 2**
- Actions Import Solution (Skip Lower Version false & Holding Solution false) and Apply Solution Upgrade in pipeline
Result: on existing solutions it doesn't do the upgrade (but an update). On new solutions it does the import but fails the upgrade (of course, because there is no upgradeable layer).

**Example 3**
- Actions Import Solution (Skip Lower Version true & Holding Solution true) and Apply Solution Upgrade in pipeline
Result: fails on new solutions (see also #499). Works on existing solutions with a higher version number. Solutions with a lower version number than installed are not imported, BUT the subsequent upgrade step fails.

As you can see, I can't seem to figure out a combination of settings that will ALWAYS work (regardless if a solution is new or pre-existing and with a higher or lower version number). There is always a step that fails.

How should I configure this pipeline? And if not possible, can you build something in the actions to make this possible?

**Comment by jcoxdco** [comment]
@YannickRe 
You may want to upvote the discussion here: Other have run into this too. :)
https://github.com/microsoft/powerplatform-build-tools/discussions/385
